tests/speed_change/test_o8c48_speed_change.py

```diff
@@ -0,0 +1,173 @@
+    duthost.shell("config load_minigraph -y")
+    time.sleep(120)
+    conn_facts = conn_graph_facts['device_conn'][duthost.hostname]
+    peer_device = conn_facts['Ethernet96']['peerdevice']
+    peerdev_ans = fanouthosts[peer_device]
+    fanout_os = peerdev_ans.get_fanout_os()
+
+    mg_facts = duthost.get_extended_minigraph_facts(tbinfo)
+    etp = list(mg_facts['minigraph_ports'].values())
+
+
+# Get DUT Ethernet interface names for desired aliases
+    dut_interfaces = []
+    for interface in etp:
+        if interface['alias'] in etp_interfaces:
+            dut_interfaces.append(interface['name'])
+
+    peer_ports = []  # List to store peer ports for each interface
+
+    for interface in dut_interfaces:
+        if interface in conn_facts:
+            peer_device = conn_facts[interface]['peerdevice']
+            peer_port = conn_facts[interface]['peerport']
+            peer_ports.append(peer_port)
+            print("DUT Interface: {}, Peer Device: {}, Peer Port: {}".format(interface, peer_device, peer_port))
+        else:
+            print("DUT Interface {} not found in connection facts".format(interface))
+
+
+# check DUT speed after speed change and compare with existing fanout interface speed
+    out = duthost.show_and_parse("show interface status {}".format(dut_interfaces[0]))
+    dut_speed = out[0]['speed']
+#    output = peerdev_ans.host.command("show interface status Ethernet248")
+    output = peerdev_ans.host.command("show interface status {}".format(peer_ports[0]))
+    output = str(output)
+    lines = output.splitlines()
+    speed_line = lines[0]
+    speed_match = re.search(r"\d+G", speed_line)
+    if speed_match:
+        fanout_speed = speed_match.group(0)
+        print("Speed:", fanout_speed)
+    else:
+        print("Speed not found.")
+
+#   check if speed of DUT after modification and the existing fanout interface speed are notsame
+    if dut_speed != fanout_speed:
+        if fanout_os == "sonic":
+            peerdev_ans.host.command("sudo cp /etc/sonic/config_db.json /etc/sonic/config_db_work.json")
+        try:
+
+            result = peerdev_ans.host.command("sudo cat /etc/sonic/config_db.json")
+            config_db_json = result["stdout"]
+            config_db = json.loads(config_db_json)
+
+            for peer_port in peer_ports:
+                show_intf_output = peerdev_ans.host.command("show interface status {}".format(peer_port))
+                speed_match = re.search(r"\s(\d+)G\s", show_intf_output["stdout"])
+                speed = int(speed_match.group(1)) if speed_match else None
+
+                if speed == 400:
+                    config_db["PORT"][peer_port]["lanes"] = ",".join(config_db["PORT"][peer_port]["lanes"]
+                                                                     .split(",")[:-4])
+                    config_db["PORT"][peer_port]["speed"] = "100000"
+                    config_db["PORT"][peer_port]["fec"] = "rs"
```

Comment:
In my experiment, need to remove "fec" field when change fanout port speeed to 100g.
wondering if can always keep "fec" value as "rs" in your test.